json_object.h

```diff
@@ -899,6 +899,28 @@ extern int json_object_set_string_len(json_object* obj, const char* new_value, i
 extern int json_object_equal(struct json_object *obj1,
 			     struct json_object *obj2);
 
+/**
+ * Copy the contents of the JSON object.
+ * If the destination object is NULL, it will be allocated.
+ * If the destination object is non-NULL, it will be overwrriten
```

Comment:
How about:
> If *dst is non-NULL, it will be passed to json_object_put (which may free it if the refcount drops to 0)
> and a completely new object will be created and assigned to *dst.